presto-main/src/main/java/com/facebook/presto/operator/scalar/MapConstructor.java

```diff
@@ -101,7 +111,19 @@ public static Block createMap(MapType mapType, Block keyBlock, Block valueBlock)
             mapType.getKeyType().appendTo(keyBlock, i, blockBuilder);
             mapType.getValueType().appendTo(valueBlock, i, blockBuilder);
         }
+        mapBlockBuilder.closeEntry();
+        pageBuilder.declarePosition();
+
+        return mapType.getObject(mapBlockBuilder, mapBlockBuilder.getPositionCount() - 1);
+    }
 
-        return blockBuilder.build();
+    public static class State
+    {
+        PageBuilder pageBuilder;
```

Comment:
Made it `final`. This needs to be `public` because its used in generated code and the lookup is not constructed in this class.